### PR TITLE
Update devel/dparser to version 1.31 and fix https://trac.macports.or…

### PR DESCRIPTION
**Pull request by jplevyak**
…g/ticket/57861

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.x
Xcode 8.x

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

File: devel/dparser/Portfile [hunk 1]
```diff
@@ -19,14 +19,15 @@ long_description \
     just well formed computer languages and data files, but just about any \
     wacky situation that occurs in the real world.
 
-homepage            http://dparser.sourceforge.net/
-master_sites        sourceforge
-checksums           rmd160  5075cede706ee98c4c7bec8288f4ef919bf1dc97 \
-                    sha256  4c4a61bd5954d75da89cbcfe8f6aac3a113769939b31ce2bf4b20b7b3a31cdd8
-distname            d-${version}-src
+homepage            http://github.com/jplevyak/dparser
```
Comment (pmetzger) [inline devel/dparser/Portfile]: Since the source is now on Github, the Github portgroup should be used; it will automate most of the work.

Comment (pmetzger) [reply devel/dparser/Portfile]: FYI, I believe "https:" is the correct URL protocol these days. But that could be fixed in a future update.

Comment (pmetzger) [reply devel/dparser/Portfile]: Oh, and also, it isn't needed at all, what was I thinking, the portgroup handles it already by default. If I commit this now, please submit a PR to fix it later.

File: devel/dparser/Portfile [hunk 2]
```diff
@@ -19,22 +20,18 @@ long_description \
...
+master_sites        https://github.com/jplevyak/dparser/releases/download/v${version}
```
Comment (pmetzger) [inline devel/dparser/Portfile]: Yah, and the github portgroup handles this, too. I suggest that you fix this before I merge it. Sorry for the delays.

File: devel/dparser/Portfile [hunk 3]
```diff
@@ -19,22 +20,18 @@ long_description \
...
+checksums           rmd160  0986a5cb1951d240aadf0eb1b659b4909ede3703 \
+                    sha256  40e5f262b2ac24895c9fa232100c825663f64e674d34122991432efdcd7849b0 \
+                    size 186818
+distname            dparser-${version}-src
```
Comment (pmetzger) [inline devel/dparser/Portfile]: distname should be handled using the portgroup as well...

File: devel/dparser/Portfile [hunk 4]
```diff
@@ -19,22 +20,18 @@ long_description \
...
 patchfiles          patch-Makefile
 
-worksrcdir          d
+worksrcdir          dparser
```
Comment (pmetzger) [inline devel/dparser/Portfile]: ...And so should worksrcdir...

**Comment by pmetzger** [comment]
Howdy! So sorry for the delay, I wasn't reviewing changes for a few days. A bunch of the things you still have in the Portfile are handled by the portgroup so they can be removed.

**Comment by kencu** [comment]
please excuse me. I could see it's been a couple of weeks, and I had a few minutes to fix up the issues @pmetzger raised for you. I know it gets a bit tedious.

I hope it's all to your liking, and I apologize for force-pushing to your repo to get this moving. If the CI systems builds it all, I think we can go ahead and get this merged.

I took the liberty of enabling tests and also doing some standard MacPorts stuff for universal builds.

**Comment by kencu** [comment]
OK. We have a clean sweep of success on the CI system, and here for me, including universal builds, and including tests. Looks good to go.

**Pull request merged**